core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -374,17 +349,35 @@ class DAGScheduler(
   }
 
   /**
+   * Create a ResultStage associated with the provided jobId.
+   */
+  private def createResultStage(
+      rdd: RDD[_],
+      func: (TaskContext, Iterator[_]) => _,
+      partitions: Array[Int],
+      jobId: Int,
+      callSite: CallSite): ResultStage = {
+    val id = nextStageId.getAndIncrement()
+    val stage = new ResultStage(
+      id, rdd, func, partitions, getOrCreateParentStages(rdd, jobId), jobId, callSite)
```

Comment:
`getOrCreateParentStages` should be called before getting the id for the result stage, otherwise the result stage will get numbered below the dependent stages.